client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/QueryDSLDocumentationTests.java

```diff
@@ -249,6 +250,12 @@ public void testMatchAll() {
         // end::match_all
     }
 
+    public void testMatchNone() {
+        // tag::match_none
```

Comment:
These tests (and the tag itself) are really only useful in the context of the documentation where we extract example java snippets from the code. Take a look at https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-query-dsl-match-all-query.html and the corresponding documentation file https://github.com/elastic/elasticsearch/blob/6.2/docs/java-api/query-dsl/match-all-query.asciidoc to see what I mean.

In the query DSL docs at https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-match-all-query.html we group "match_all" and "match_none" on one page for convenience, I'd suggest doint that for the java api docs as well. Just adding another few lines to the match-all-query.asciidoc should be enough I think.